ui/src/views/infra/HostEnableDisable.vue

```
@@ -97,37 +94,44 @@ export default {
       })
     },
     handleSubmit (e) {
-      e.preventDefault()
-      this.formRef.value.validate().then(() => {
+      if (e) {
+        e.preventDefault()
+      }
+      this.$refs.formRef.validate().then(() => {
         const values = toRaw(this.form)
-
-        var data = {
+        const data = {
           allocationstate: this.allocationstate,
           id: this.resource.id
         }
         if (values.reason) {
           data.annotation = values.reason
         }
-        api('updateHost', data).then(_ => {
-          this.$emit('close-action')
-        })
+        api('updateHost', data)
+          .then(response => {
+            this.$emit('refresh-data')
+            this.$emit('close-action')
+          })
+          .catch(error => {
+            console.error('Error updating host:', error)
+            this.$emit('close-action')
+          })
+      }).catch(error => {
+        console.error('Validation failed:', error)
```

It would be better to return a notification error in the UI instead of printing the error to the console

---

Sure, did it to test the development only.